spring-xd-dirt/src/main/java/org/springframework/xd/dirt/rest/StreamsController.java

```diff
@@ -67,4 +75,27 @@ public StreamsController(StreamDeployer streamDeployer, StreamDefinitionReposito
 	protected StreamDefinition createDefinition(String name, String definition) {
 		return new StreamDefinition(name, definition);
 	}
+
+	@ResponseBody
+	@RequestMapping(value = "/clean/{transport}/{stream}")
+	@ResponseStatus(HttpStatus.OK)
+	public List<String> clean(@PathVariable String transport, @PathVariable String stream,
+			@RequestParam(required = false) String adminUri,
+			@RequestParam(required = false) String user,
+			@RequestParam(required = false) String pw,
+			@RequestParam(required = false) String vhost,
+			@RequestParam(required = false) String busPrefix) {
+		Assert.isTrue("rabbit".equals(transport), "Only rabbit transports can be cleaned");
+		String adminUri2 = adminUri == null ? "http://localhost:15672" : adminUri;
+		String user2 = user == null ? "guest" : user;
+		String pw2 = pw == null ? "guest" : pw;
+		String vhost2 = vhost == null ? "/" : vhost;
+		String busPrefix2 = busPrefix == null ? "xdbus." : busPrefix;
+		List<String> results = busCleaner.clean(adminUri2, user2, pw2, vhost2, busPrefix2, stream);
+		if (results.size() == 0) {
+			results.add("No queues named " + busPrefix2 + stream + ".* found");
```

Comment:
Should not convey this kind of result the same way a normal result is conveyed. Different HTTP code seems like a better option